tasks/main.yml

```diff
@@ -36,3 +39,9 @@
     dest: /etc/profile.d/java_home.sh
     mode: 0644
   when: java_home is defined and java_home != ''
+
+- name: check java version
+  command: java -version
+  changed_when: false
+  register: javaversion
+- debug: var=javaversion
```

Comment:
there is more discussion here.
I'm usually using
```
include_vars: "{{ ansible_distribution }}-{{ ansible_distribution_version }}.yml"
```
but no good for centos/RHEL
```
include_vars: "RedHat-{{ ansible_distribution_version.split('.')[0] }}.yml"
```
if using ansible_distribution here, it means differentiating RHEL and Centos with identitical content.
As for the split, ansible 2.x introduces ansible_distribution_major_version which might be good for all but not sure. My usual tests are Ubuntu LTS Xenial+Trusty and centos7, sometimes older and debian8.
For now, I left as is on my roles. what is your preferences.

For the debug, agree. but a few times I get caught by having the wrong java and couldn't figure which one without verbose mode or connecting the box :)